src/test/ui/layout/debug.rs

```diff
@@ -0,0 +1,7 @@
+#![feature(never_type, rustc_attrs)]
+#![crate_type = "lib"]
+
+enum E { Foo, Bar(!, i32, i32) }
+
+#[rustc_layout(debug)]
+type Test = E; //~ ERROR: layout debugging
```

Comment:
We should allow it on other type definitions too, to avoid having to use a type alias.

Reply:
I added that to https://github.com/rust-lang/rust/issues/69852. It seems orthogonal to adding the debug printing.